flaky test: TestMemKVClient

```
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1

goroutine 106060 [running]:
testing.tRunner.func1.2({0x180b300, 0xc001d1e078})
	/opt/hostedtoolcache/go/1.19.10/x64/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.19.10/x64/src/testing/testing.go:1399 +0x39f
panic({0x180b300, 0xc001d1e078})
	/opt/hostedtoolcache/go/1.19.10/x64/src/runtime/panic.go:884 +0x212
github.com/google/btree.(*node[...]).iterate(0xc0012eb3c0, 0x1b30b80, {{0x1b30b80?, 0xc001e5c1b0?}, 0xb0?}, {{0x0?, 0x0?}, 0xb0?}, 0x1, 0x1, ...)
	/home/runner/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:522 +0x4c7
github.com/google/btree.(*node[...]).iterate(0xc0012fa400, 0x1b30b80, {{0x1b30b80?, 0xc001e5c1b0?}, 0x0?}, {{0x0?, 0x0?}, 0x20?}, 0x1, 0x1, ...)
	/home/runner/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:527 +0x49b
github.com/google/btree.(*node[...]).iterate(0xc0012fa440, 0x0, {{0x1b30b80?, 0xc001e5c1b0?}, 0xc7?}, {{0x0?, 0x0?}, 0x1?}, 0x1, 0x0, ...)
	/home/runner/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:527 +0x49b
github.com/google/btree.(*BTreeG[...]).AscendGreaterOrEqual(0x1?, {0x1b30b80?, 0xc001e5c1b0?}, 0xc001723be0?)
	/home/runner/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:770 +0x58
github.com/google/btree.(*BTree).AscendGreaterOrEqual(...)
	/home/runner/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:1023
github.com/juicedata/juicefs/pkg/meta.(*memKV).scan(0xc001563c50?, {0xc001bc8000?, 0x1?, 0x1?}, 0xc001e5c180?)
	/home/runner/work/juicefs/juicefs/pkg/meta/tkv_mem.go:242 +0x1e5
github.com/juicedata/juicefs/pkg/meta.(*kvMeta).scanAllChunks(0xc0010f1300, {0xf4240?, 0x0?}, 0xc002eb2000, 0xc000f53c80)
	/home/runner/work/juicefs/juicefs/pkg/meta/tkv.go:2477 +0x117
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).CompactAll(0xc000062f00, {0x1b4fc50, 0xc00007f100}, 0x8, 0xc000f53c80)
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:2157 +0x194
github.com/juicedata/juicefs/pkg/meta.testCompaction(0xc000aca[340](https://github.com/juicedata/juicefs/actions/runs/5309546841/jobs/9610334713#step:8:341), {0x1b5c910?, 0xc0010f1300}, 0x0)
	/home/runner/work/juicefs/juicefs/pkg/meta/base_test.go:1158 +0xbdb
github.com/juicedata/juicefs/pkg/meta.testMeta(0xc000aca340, {0x1b5c910, 0xc0010f1300})
	/home/runner/work/juicefs/juicefs/pkg/meta/base_test.go:130 +0x145
github.com/juicedata/juicefs/pkg/meta.TestMemKVClient(0xc000aca340)
	/home/runner/work/juicefs/juicefs/pkg/meta/tkv_test.go:35 +0x15f
testing.tRunner(0xc000aca340, 0x197c7d0)
	/opt/hostedtoolcache/go/1.19.10/x64/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.19.10/x64/src/testing/testing.go:1493 +0x35f
FAIL	github.com/juicedata/juicefs/pkg/meta	126.868s
```

https://github.com/juicedata/juicefs/actions/runs/5309546841/jobs/9610334713

--- a/pkg/meta/tkv_mem.go
+++ b/pkg/meta/tkv_mem.go
@@ -236,17 +236,16 @@ func (c *memKV) txn(f func(*kvTxn) error, retry int) error {
 
 func (c *memKV) scan(prefix []byte, handler func(key []byte, value []byte)) error {
 	c.Lock()
-	defer c.Unlock()
+	snap := c.items.Clone()
+	c.Unlock()
 	begin := string(prefix)
 	end := string(nextKey(prefix))
-	c.items.AscendGreaterOrEqual(&kvItem{key: begin}, func(i btree.Item) bool {
+	snap.AscendGreaterOrEqual(&kvItem{key: begin}, func(i btree.Item) bool {
 		it := i.(*kvItem)
 		if end != "" && it.key >= end {
 			return false
 		}
-		c.Unlock()
 		handler([]byte(it.key), it.value)
-		c.Lock()
 		return true
 	})
 	return nil

